### PR TITLE
Feat/improve toast description

### PR DESCRIPTION
**Pull request by ErikssonJoakim**
This PR implements #375

### CODE REVIEW

File: stories/atoms/components/toast/Toast.stories.mdx [hunk 1]
```diff
@@ -45,7 +47,7 @@ Several severity levels make it possible to fully configure the type of message
 
 <Story
   name="Overview"
-  argTypes={{ isOpened: { control: false } }}
+  argTypes={{ isOpened: { control: false }, description: { control: false } }}
```
Comment (ccamel) [inline stories/atoms/components/toast/Toast.stories.mdx:50]: It's a disappointment that you can't control that property. Why not force the control as a string to be able to play anyway?

File: src/ui/atoms/toast/Toast.tsx [hunk 2]
```diff
@@ -62,9 +63,13 @@ export const Toast: React.FC<ToastProps> = ({
         )}
         {description && (
           <ToastPrimitive.Description asChild className="okp4-toast-description">
-            <Typography as="span" color="highlighted-text" fontSize="small" fontWeight="light">
-              {description}
-            </Typography>
+            {typeof description === 'string' ? (
```
Comment (ccamel) [inline src/ui/atoms/toast/Toast.tsx:66]: I'd like to see this condition wrapped into a `typeof` guard function.

File: stories/atoms/components/toast/Toast.stories.mdx [hunk 3]
```diff
@@ -316,6 +321,62 @@ These two elements are optional and are not inter-related.
               severityLevel="success"
             />
           </div>
+          <div>
+            <Button
+              label="Bot Anik"
```
Comment (fredericvilcot) [inline stories/atoms/components/toast/Toast.stories.mdx:326]: Please, add a better label for the toast button that will be more explicit for the user

File: stories/atoms/components/toast/Toast.stories.mdx [hunk 4]
```diff
@@ -316,6 +321,62 @@ These two elements are optional and are not inter-related.
...
+                  <Typography
+                    as="span"
+                    color="highlighted-text"
+                    fontSize="small"
+                    fontWeight="light"
+                  >
+                    Bot Anik is a bot like no other, she'll kick you around without any trouble! She
+                    gets rid of all who messes around, no one can beat our bot! 🎵
+                  </Typography>
+                  <Typography
```
Comment (fredericvilcot) [inline stories/atoms/components/toast/Toast.stories.mdx:343]: If you don't mind, I'd prefer the use of the `Button` component instead of a custom element

**Review by fredericvilcot — CHANGES_REQUESTED** [review]

File: stories/atoms/components/toast/Toast.stories.mdx (hunk 1 above)
Comment (ErikssonJoakim) [reply stories/atoms/components/toast/Toast.stories.mdx]: Great catch, I didn't know we could do that with storybook

File: src/ui/atoms/toast/Toast.tsx (hunk 2 above)
Comment (ErikssonJoakim) [reply src/ui/atoms/toast/Toast.tsx]: I've put the guard function in utils as it is a general purpose function

File: stories/atoms/components/toast/Toast.stories.mdx (hunk 3 above)
Comment (ErikssonJoakim) [reply stories/atoms/components/toast/Toast.stories.mdx]: Good point! 👍

**Review by fredericvilcot — APPROVED** [review]

**Review by ccamel — APPROVED** [review]
👍

**Review by JoeyDjr — APPROVED** [review]

**Review by lolottetheclash — CHANGES_REQUESTED** [review]
I think we can remove this part of code: `export const ToastComponent = ({ ... })` because it's not used.

And maybe we can change :
`const handleState = (name, isOpened) => () => {
      setState({ name: name, isOpened: isOpened })
}`
into  :
`const handleState = (name, isOpened) => () => {
      setState({ name, isOpened })
}`

What do you think?

**Comment by ErikssonJoakim** [comment]
@lolottetheclash Great observations! I'd say it's safe to remove those operations

**Review by lolottetheclash — APPROVED** [review]
Great !

**Review by JoeyDjr — APPROVED** [review]

**Pull request merged**